### DISCUSSION TITLE
`ui.timer` doesn't work as expected

### DISCUSSION THREAD

**Discussion by amks1** [zauberzeug/nicegui#2778] [Q&A]
### Description

I'm trying to create a native application as follows:

```python3
from nicegui import ui
import time, asyncio

@ui.page('/next_page')
def next_page():
    ui.label("This is the next page!")


def check_something_from_db():
    print("Checking something...")
    time.sleep(5)
    print("Done, going to next page!")
    ui.navigate.to(next_page)


with ui.column() as spinner:
    ui.label(f'Doing something without human input...')
    ui.spinner(size='2em')
ui.timer(2, lambda: check_something_from_db(), once=True)    

ui.run(
    reload=False,
    native=True,
)
```

The spinner column is not shown at all. There's a blank page and then it shows the "This is the next page!" text.
Console Output:
```
Checking something...
Done, going to next page!
```

**Comment by falkoschindler** [comment]
Hi @amks1,

Your example is working on my machine. But blocking the thread synchronously for 5 seconds is not recommended and you should use `asyncio.sleep` or `run.cpu_bound`, respectively (see https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-long-running-function-blocking-ui-updates). This might resolve your problem.
